engines/private/private.h

```diff
@@ -0,0 +1,300 @@
+    Audio::SoundHandle _bgSoundHandle;
+    Video::SmackerDecoder *_videoDecoder;
+    Common::InstallerArchive _installerArchive;
+
+    Common::Error run() override;
+    void restartGame();
+    void clearAreas();
+    void initializePath(const Common::FSNode &gamePath) override;
+
+    // User input
+
+    void selectPauseMovie(Common::Point);
+    void selectMask(Common::Point);
+    void selectExit(Common::Point);
+    void selectLoadGame(Common::Point);
+    void selectSaveGame(Common::Point);
+
+    // Cursors
+
+    bool cursorPauseMovie(Common::Point);
+    bool cursorExit(Common::Point);
+    bool cursorMask(Common::Point);
+
+    bool hasFeature(EngineFeature f) const override;
+    bool canLoadGameStateCurrently() override {
+        return true;
+    }
+    bool canSaveAutosaveCurrently() override  {
+        return false;
+    }
+    bool canSaveGameStateCurrently() override {
+        return true;
+    }
+
+    Common::Error loadGameStream(Common::SeekableReadStream *stream) override;
+    Common::Error saveGameStream(Common::WriteStream *stream, bool isAutosave = false) override;
+    void syncGameStream(Common::Serializer &s);
+
+    Common::String convertPath(Common::String);
+    void playVideo(const Common::String &);
+    void skipVideo();
+
+    void loadImage(const Common::String &file, int x, int y);
+    void drawScreenFrame(Graphics::Surface *);
+
+    void changeCursor(Common::String);
+    void initCursors();
+
+    Graphics::ManagedSurface *loadMask(const Common::String &, int, int, bool);
+    void drawMask(Graphics::ManagedSurface *);
+    bool inMask(Graphics::ManagedSurface*, Common::Point);
+    uint32 _transparentColor;
+    void drawScreen();
+
+    // global state
+    const Common::Point *_origin;
+    Common::String *_nextSetting;
+    Common::String *_currentSetting;
+    Common::String *_nextVS;
+    Common::String *_frame;
+    bool            _toTake;
+
+    // Dossiers
+
+    DossierArray _dossiers;
+    unsigned int _dossierSuspect;
+    unsigned int _dossierPage;
+    MaskInfo *_dossierNextSuspectMask;
+    MaskInfo *_dossierPrevSuspectMask;
+    bool selectDossierNextSuspect(Common::Point);
+    bool selectDossierPrevSuspect(Common::Point);
+    void loadDossier();
+
+    // Police Bust
+
+    int policeVideoIndex;
+    void policeBust();
+    bool _policeBustEnabled;
+    void startPoliceBust();
+    void checkPoliceBust();
+    int _numberClicks;
+    int _maxNumberClicks;
+    int _sirenWarning;
+    Common::String *_policeBustSetting;
+
+    // Diary
+
+    InvList inventory;
+    Common::String *_diaryLocPrefix;
+    void loadLocations(Common::Rect *);
+    void loadInventory(uint32, Common::Rect *, Common::Rect *);
+
+    // Save/Load games
+    MaskInfo *_saveGameMask;
+    MaskInfo *_loadGameMask;
+
+    int _mode;
+    bool _modified;
+    Common::String *_nextMovie;
+    PlayedMediaTable _playedMovies;
+    PlayedMediaTable _playedPhoneClips;
+    Common::String *_repeatedMovieExit;
```

Comment:
Why do you keep pointers here, with `Common::String`? Wouldn't it be better using the objects directly?

Reply:
It's useful to have `_nextMovie` and `_repeatedMovieExit` to know when they are not assigned to anything (e.g. NULL).